strapi/strapi
# fix: add user-agent to patreon oauth requests

### What does it do?

The users & permissions plugin has a provider for logging in via Patreon. Recently requests to Patreon started to fail with a 400 error. It appears to be because of the lack of a `User-Agent` header in the requests. This PR adds the `User-Agent` header similar to the existing GitHub provider.

See a related discussion including Patreon's response at https://www.patreondevelopers.com/t/status-400-https-www-patreon-com-api-oauth2-v2-identity/8836/9.

### Why is it needed?

Without this PR, logging in via Patreon always fails.

### How to test it?

1. Get set up on Patreon as a creator and create a client on https://www.patreon.com/portal.
2. In the Users & Permissions plugin, enable the Patreon provider and configure the client based on the info from Patreon.
3. Set up a user in Strapi that has the same email address as a subscriber on your Patreon creator account.
4. Create a simple front end app that Patreon redirects back to. The front end app needs to send a `GET` request to `api/auth/patreon/callback` with the data that Patreon sends back in the URL redirection. This request should return the correct user.

### Related issue(s)/PR(s)

N/A
